core/lib/multivm/src/versions/vm_fast/vm.rs

```diff
@@ -145,14 +146,15 @@ impl<S: ReadStorage> Vm<S> {
                             .as_u64();
 
                         let pubdata_published = self.inner.world_diff.pubdata.0 as u32;
+                        //pubdata_published -= pubdata_before;
```

Comment:
Please remove if this code isn't needed. BTW, how would pubdata computations work if the entire batch is executed (i.e., `execution_mode == VmExecutionMode::Batch`)? I'd think that `pubdata_before` needs to be updated somewhere in `run()` (in `TxHasEnded` hook handler?).

Reply:
I've changed it to be updated here on the spot, as for now i do not have a better theory about where it should be updated.

Reply:
I don't think it's correct (it would probably lead to the previous issue of `pubdata_published` for a transaction including pubdata produced by the bootloader / system contracts after executing the previous transaction), but let's fix this sequentially.